core/handler/manager_server.go

```diff
@@ -266,8 +268,8 @@ func (h *handlerManager) GetDevicesForApplication(ctx context.Context, in *pb.Ap
 }
 
 func (h *handlerManager) GetApplication(ctx context.Context, in *pb.ApplicationIdentifier) (*pb.Application, error) {
-	if !in.Validate() {
-		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Application Identifier", "validation failed"))
+	if err := in.Validate(); err != nil {
+		return nil, errors.NewErrInvalidArgument("Application Identifier", err.Error())
```

Comment:
Be consistent with `errors.Wrap` vs wrapping it in `errors.NewErrInvalidArgument`